File: ABB-setup/dv-processing-rel_1.7/docs/source/assets/code_samples/python/reading_data/sample4.py
import dv_processing as dv
import cv2 as cv

# Open any camera
capture = dv.io.CameraCapture()

# Initiate a preview window
cv.namedWindow("Preview", cv.WINDOW_NORMAL)

# Run the loop while camera is still connected
while capture.isRunning():
    # Read a frame from the camera
    frame = capture.getNextFrame()

    # The method does not wait for frame arrive, it returns immediately with
    # latest available frame or if no data is available, returns a `None`
    if frame is not None:
        # Print received packet time range
        print(f"Received a frame at time [{frame.timestamp}]")

        # Show a preview of the image
        cv.imshow("Preview", frame.image)
    cv.waitKey(2)

File: source_code/ABB-setup/dv-processing-rel_1.7/docs/source/assets/code_samples/c++/reading_data/sample4.cpp
#include <dv-processing/io/camera_capture.hpp>

#include <opencv2/highgui.hpp>

int main() {
	// Open any camera
	dv::io::CameraCapture capture;

	// Initiate a preview window
	cv::namedWindow("Preview", cv::WINDOW_NORMAL);

	// Run the loop while camera is still connected
	while (capture.isRunning()) {
		// Read a frame, check whether it is correct.
		// The method does not wait for frame arrive, it returns immediately with
		// the latest available frame or if no data is available, returns a `std::nullopt`.
		if (const auto frame = capture.getNextFrame(); frame.has_value()) {
			std::cout << *frame << std::endl;

			// Show a preview of the image
			cv::imshow("Preview", frame->image);
		}
		cv::waitKey(2);
	}

	return 0;
}
